src/main/java/org/jenkinsci/plugins/github_branch_source/TagDiscoveryTrait.java

```diff
@@ -83,6 +84,7 @@ public boolean includeCategory(@NonNull SCMHeadCategory category) {
     /**
      * Our descriptor.
      */
+    @Symbol("gitHubTagDiscovery")
```

Comment:
Really not happy that we need to prefix all the symbols in a global namespace

Reply:
This is because of how JobDSL and CasC load plugins dynamically. Given these are interfaces that can be implemented by different plugins (GitHub, GitLab, BitBucket, etc) we need to have these annotations so that there are no conflicts. When using latest master there are clashes in the names (CasC uses the class name if no Symbol annotation is found in the Descriptor), which prevents the instantiation of these instances at runtime!

Reply:
I know why the current impl of symbol cannot handle it. I am sad that we are working around by kicking the van down the road rather than fixing symbol so that it can be contextually aware

Reply:
is it related to declarative issue where `when tag XXX` conflicts with symbol from my own extension point?

Reply:
Then somebody should change javadoc. Because following this https://github.com/jenkinsci/structs-plugin/blob/master/annotation/src/main/java/org/jenkinsci/Symbol.java#L21-L22  i named my stuff and it now conflicts with some declarative stuff https://github.com/KostyaSha/github-integration-plugin/issues/274#issuecomment-396369173

Reply:
AFAIK the Javadoc remains accurate: a symbol is expected to be unique within a given `ExtensionPoint`. The issue here, IIUC, is that there is a way for calling code to infer a more specific scope than the generic Jenkins extension system knows about, and in the interests of simplicity `structs` does not support defining more specific scopes.

Reply:
Could you please add to demos directory a set of examples to demonstrate this feature ?